File: solutions/zig/space-age/1/space_age.zig
pub const Planet = enum {
    mercury,
    venus,
    earth,
    mars,
    jupiter,
    saturn,
    uranus,
    neptune,

    pub fn age(self: Planet, seconds: usize) f64 {
        const years = [8]f16{
            0.2408467,
            0.61519726,
            1.0,
            1.8808158,
            11.862615,
            29.447498,
            84.016846,
            164.79132,
        };
        const earth_seconds = 31557600;
        return @as(f64, @floatFromInt(seconds)) / earth_seconds / years[@intFromEnum(self)];
    }
};

File: solutions/go/space-age/1/space_age.go
package space

type Planet string
type EarthYear = float64

const EARTH_SECONDS = 31557600

var planets = map[Planet]EarthYear{
	"Mercury": 0.2408467,
	"Venus": 0.61519726,
	"Earth": 1.0,
	"Mars": 1.8808158,
	"Jupiter": 11.862615,
	"Saturn": 29.447498,
	"Uranus": 84.016846,
	"Neptune": 164.79132,
}

// Age computes the age in seconds in another planets time
func Age(seconds float64, planet Planet) float64 {
	return seconds / EARTH_SECONDS / planets[planet]
}
